src/crimson/os/alienstore/alien_store.cc

```diff
@@ -338,13 +338,7 @@ AlienStore::get_attrs(CollectionRef ch,
   return seastar::do_with(attrs_t{}, [=] (auto &aset) {
     return tp->submit(ch->get_cid().hash_to_shard(tp->size()), [=, &aset] {
       auto c = static_cast<AlienCollection*>(ch.get());
-      std::map<std::string, ceph::bufferptr> blueaset;
-      const auto r = store->getattrs(c->collection, oid, blueaset);
-      for (auto& [bluekey, blueval] : blueaset) {
-        ceph::bufferlist bl;
-        bl.push_back(std::move(blueval));
-        aset.emplace(std::move(bluekey), std::move(bl));
-      }
+      const auto r = store->getattrs(c->collection, oid, aset);
```

Comment:
Hmm, that's stange. IIUC `attrs_t` is still about `bufferlist` while the sole change for `BlueStore` is just about the comparator:

```diff
--- a/src/os/bluestore/BlueStore.cc
+++ b/src/os/bluestore/BlueStore.cc
@@ -10431,7 +10431,7 @@ int BlueStore::getattr(
 int BlueStore::getattrs(
   CollectionHandle &c_,
   const ghobject_t& oid,
-  map<string,bufferptr>& aset)
+  map<string,bufferptr,less<>>& aset)
 {
   Collection *c = static_cast<Collection *>(c_.get());
   dout(15) << __func__ << " " << c->cid << " " << oid << dendl;
diff --git a/src/os/bluestore/BlueStore.h b/src/os/bluestore/BlueStore.h
index a1249607a4d2a..c08cc6d61aac8 100644
--- a/src/os/bluestore/BlueStore.h
+++ b/src/os/bluestore/BlueStore.h
@@ -2836,7 +2836,7 @@ class BlueStore : public ObjectStore,
 	      ceph::buffer::ptr& value) override;
 
   int getattrs(CollectionHandle &c, const ghobject_t& oid,
-	       std::map<std::string,ceph::buffer::ptr>& aset) override;
+	       std::map<std::string,ceph::buffer::ptr, std::less<>>& aset) override;
 
   int list_collections(std::vector<coll_t>& ls) override;
 ```

If so, how would it pass the `make check` bot?

Reply:
@rzarzynski there are two variants of `getattrs()`. one for bufferptr and the other for bufferlist. before this change, the variant which returns `std::map<std::string, ceph::bufferptr>` is used. after this change the variant returning `std::map<std::string, ceph::bufferlist>` is used. so we don't need to box the bufferptr as bufferlist here manually.

Reply:
Ah, I see. Thanks.